modules/core/shared/src/main/scala/lucuma/core/enum/TwilightType.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma
+package core
+package enum
+
+import cats.implicits._
+import lucuma.core.math.Angle
+import lucuma.core.math.Declination
+import lucuma.core.math.syntax.all._
+import lucuma.core.util.Enumerated
+
+/**
+  * Definition for how the range from sunset to sunrise should be defined for
+  * a night. There are various standard options for definition where the night
+  * begins and ends which are represented as static constants in side this
+  * class.
+  */
+sealed abstract class TwilightType(val tag: String, val horizonAngle: Declination)
```

Comment:
I went with this concept because it seemed to me that it's an elevation, which is an angle that can only have values between [-90, 90] degrees, and `Declination` models that accurately. We are also using `Declination` for target/sun/moon elevation in `SkyCalcResults`.

To model `Latitude` we made an alias `type Latitude = Declination`, maybe we should define another `Elevation` alias and use it in these cases?

Reply:
Allowing for negative values, the `horizonAngle` is the Sun's altitude. At any rate the Sun will have a real declination at that time that won't be the horizon angle (though it will be related to the horizon angle through the observer's latitude).

I think I might make it an `Angle` and call it good, but if you want to investigate making `Declination` have or be a kind of more general [-90, 90] thing that might be good.  What do you think @tpolecat?